test/functional/lua/highlight_spec.lua

```diff
@@ -1,14 +1,187 @@
+local Screen = require('test.functional.ui.screen')
 local helpers = require('test.functional.helpers')(after_each)
-local exec_lua = helpers.exec_lua
+local clear = helpers.clear
+local command = helpers.command
 local eq = helpers.eq
 local eval = helpers.eval
-local command = helpers.command
-local clear = helpers.clear
-
-describe('vim.highlight.on_yank', function()
+local exec_lua = helpers.exec_lua
+local feed = helpers.feed
 
-  before_each(function()
+before_each(function()
     clear()
+end)
+
+describe('highlight.on_yank()', function()
+  local screen
+  local HL_TIMEOUT = 30
+  local enable_hl_on_yank = function(args)
+    local higroup = args['higroup'] or 'IncSearch'
```

Comment:
wonder why we introduced a new name for "highlight group" where "hl" (instead of "hi") was already the abbreviation established by `:help` (`:help hl-IncSearch` etc)

Reply:
nobody complained on the original PR ;)

Reply:
And I was following the pattern `hi[ghlight] foo guifg=...`, I guess.

Reply:
But note that the `higroup` is part of the documented API now; changing this would be breaking (unless we add compatibility shims).

Reply:
yes, understood

Reply:
(otherwise I would have changed it in a heartbeat ;))

Will keep it in mind for any future _necessary_ API change.